--- a/frameworks/native/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/frameworks/native/services/surfaceflinger/SurfaceFlinger.cpp
@@ -86,6 +86,8 @@
 #endif
 
 #define DISPLAY_COUNT       1
+#define FBIOBLANK           0x4611
+#define FB_BLANK_UNBLANK    0
 
 /*
  * DEBUG_SCREENSHOTS: set to true to check that screenshots are not all
@@ -2684,6 +2686,14 @@
         int mode) {
     ALOGD("Set power mode=%d, type=%d flinger=%p", mode, hw->getDisplayType(),
             this);
+    if (mode == 2)
+    {
+	int fd, ret;
+	fd = open("/dev/graphics/fb0",O_WRONLY);
+	ret = ioctl(fd, FBIOBLANK, FB_BLANK_UNBLANK);
+	    if (ret < 0)
+		ALOGE("Error waking up LCD: %d (%s)\n", ret, strerror(errno));
+    }
     int32_t type = hw->getDisplayType();
     int currentMode = hw->getPowerMode();
 

--- a/frameworks/native/services/surfaceflinger/SurfaceFlinger_hwc1.cpp
+++ b/frameworks/native/services/surfaceflinger/SurfaceFlinger_hwc1.cpp
@@ -95,6 +95,8 @@
 
 
 #define DISPLAY_COUNT       1
+#define FBIOBLANK           0x4611
+#define FB_BLANK_UNBLANK    0
 
 /*
  * DEBUG_SCREENSHOTS: set to true to check that screenshots are not all
@@ -2701,6 +2703,14 @@
         int mode) {
     ALOGD("Set power mode=%d, type=%d flinger=%p", mode, hw->getDisplayType(),
             this);
+    if (mode == 2)
+    {
+	int fd, ret;
+	fd = open("/dev/graphics/fb0",O_WRONLY);
+	ret = ioctl(fd, FBIOBLANK, FB_BLANK_UNBLANK);
+	    if (ret < 0)
+		ALOGE("Error waking up LCD: %d (%s)\n", ret, strerror(errno));
+    }
     int32_t type = hw->getDisplayType();
     int currentMode = hw->getPowerMode();